transformer_engine/pytorch/graph.py

```
@@ -91,6 +91,9 @@ def _make_graphed_callables(
         sample_args = (sample_args,)
         sample_kwargs = (sample_kwargs,)
 
+    # Check training/inference
+    is_training = all(c.training for c in callables)
```

Is `all` the right function here or should it be `any` (or maybe we should return an error if there are some that training and some that are not)?

---

I added an assert for the mixed training/inference case, because we haven't tested it. We can remove the assert if it's proven that the mixed case is fine with `make_graphed_callables`.